oath-htb-validator.js

```diff
@@ -41,56 +41,37 @@ var partnerValidator = function (configs) {
-        properties: {
-            region: {
-                type: 'string',
-                exec: function (schema, post) {
-                    if (!validRegions.hasOwnProperty(post)) {
-                        this.report('region must be one of the predefined values: ' + Object.keys(validRegions));
-                    }
-                }
-            },
-            networkId: {
-                type: 'string',
-                minLength: 1
-            },
+        type: 'object', properties: {
             xSlots: {
-                type: 'object',
-                properties: {
+                type: 'object', properties: {
                     '*': {
-                        type: 'object',
-                        exec: function(schema, post){
-                            if (!((post.hasOwnProperty('dcn') && post.hasOwnProperty('pos')) || post.hasOwnProperty('placementId'))){
-                                this.report("xSlots must have either dcn and pos, or placementId");
+                        type: 'object', exec: function (schema, post) {
+                            if (!((post.hasOwnProperty('dcn') && post.hasOwnProperty('pos')) ||
+                                    post.hasOwnProperty('networkId') && post.hasOwnProperty('placementId'))) {
+                                this.report("xSlots must have either dcn and pos, or networkId and placementId");
                             }
                             return post;
-                        },
-                        properties: {
-                            placementId: {
-                                optional: true,
-                                type: ['string'],
-                                minLength: 1
-                            },
-                            dcn: {
-                                optional: true,
-                                type: ['string'],
-                                minLength: 1
-                            },
-                            pos: {
-                                optional: true,
-                                type: ['string'],
-                                minLength: 1
-                            },
-                            sizeId: {
-                                optional: true,
-                                type: ['string'],
-                                minLength: 1
-                            },
-                            pageId: {
+                        }, properties: {
+                            region: {
                                 optional: true,
-                                type: ['string'],
-                                minLength: 1
+                                type: 'string',
+                                exec: function (schema, post) {
+                                    if (!validRegions.hasOwnProperty(post)) {
+                                        this.report('region must be one of the predefined values: ' + Object.keys(validRegions));
```

Comment:
This should check if region has been defined in the config otherwise it will always trigger `this.report`

Reply:
Fixed